test/core.test.js

```diff
@@ -0,0 +1,47 @@
+
+const { calculateCharacterWeight } = require('../core')
+
+test('dragonborn weight in range', () => {
+    expect(calculateCharacterWeight('dragonborn')).toBeGreaterThanOrEqual(179);
+    expect(calculateCharacterWeight('dragonborn')).toBeLessThanOrEqual(367);
+});
+
+test('dwarf weight in range', () => {
+    expect(calculateCharacterWeight('dwarf')).toBeGreaterThanOrEqual(119);
+    expect(calculateCharacterWeight('dwarf')).toBeLessThanOrEqual(211);
+});
+
+test('elf weight in range', () => {
+    expect(calculateCharacterWeight('elf')).toBeGreaterThanOrEqual(92);
+    expect(calculateCharacterWeight('elf')).toBeLessThanOrEqual(170);
+});
+
+test('gnome weight in range', () => {
+    expect(calculateCharacterWeight('gnome')).toBeGreaterThanOrEqual(37);
+    expect(calculateCharacterWeight('gnome')).toBeLessThanOrEqual(43);
+});
+
+test('half-elf weight in range', () => {
+    expect(calculateCharacterWeight('half-elf')).toBeGreaterThanOrEqual(114);
+    expect(calculateCharacterWeight('half-elf')).toBeLessThanOrEqual(238);
+});
+
+test('half-orc weight in range', () => {
+    expect(calculateCharacterWeight('half-orc')).toBeGreaterThanOrEqual(144);
+    expect(calculateCharacterWeight('half-orc')).toBeLessThanOrEqual(380);
+});
+
+test('human weight in range', () => {
+    expect(calculateCharacterWeight('human')).toBeGreaterThanOrEqual(114);
+    expect(calculateCharacterWeight('human')).toBeLessThanOrEqual(270);
+});
+
+test('halfling weight in range', () => {
+    expect(calculateCharacterWeight('halfling')).toBeGreaterThanOrEqual(37);
+    expect(calculateCharacterWeight('halfling')).toBeLessThanOrEqual(43);
+});
+
+test('tiefling weight in range', () => {
+    expect(calculateCharacterWeight('tiefling')).toBeGreaterThanOrEqual(114);
+    expect(calculateCharacterWeight('tiefling')).toBeLessThanOrEqual(238);
+});
```

Comment:
These are great, nice to see how the tests work :) however I think they are going to require a bit of tweaking.  

Using a Dwarf as an example:

The max weight of 211 is never going to be reached, even if both d6's roll max:  115 + 12 = 127

The minimum weight can also fail if the result of the 2 x d6 rolls add up to less than 3 (1 - 1 / 2 - 1).

The issue is with the way you have calculated the characters weight, and to be fair the table you linked is quite misleading! I'll leave a comment on the calculateCharacterWeight() function that explains why these numbers don't match up.

Reply:
This is a doubt I had when checking the table [here](https://www.reddit.com/r/DnD/comments/abebq0/chart_with_heightweight_ranges_and_averages_for/)

It says the weight range, these are the numbers I used in the tests. I noticed that there were something wrong with the calculations but I did not want to change your original calculations. If you know the correct ones please send me a link and I will update the pull request.